Timestamp support

Robpol86 opened the pull request:
Adding optional parameter `:timestamp:` for both YouTube and Vimeo
videos.

Fixes https://github.com/sphinx-contrib/youtube/issues/17

Robpol86 committed:
Timestamp support

Adding optional parameter `:timestamp:` for both YouTube and Vimeo
videos.

Fixes https://github.com/sphinx-contrib/youtube/issues/17

Robpol86 commented:
Hmm the youtube part doesn't seem to work actually. I was fooled by YouTube since it caches the last time I viewed the video. Tested in incognito mode and it seems to ignore timestamping. Gonna close this for now until I can figure something out.

Robpol86 committed:
Fix youtube support

Robpol86 commented:
Fixed, need to use "start" instead of "t".

dham reviewed (changes requested):
Thanks for this nice contribution. I just have two concerns. One is that the timestamp seems to be dropped in LaTeX output, and the other is  what would happen if someone wanted to add more of the URL parameters that YouTube and/or Vimeo support.

I think that both of these can be fixed fairly easily. The first thing to do is to lift the processing of `:timestamp:` into a URL parameter from the html handler up to `Video.run`. This would enable this code to be shared by different backends (because it's not backend-specific). Then, the argument to `_node` could just be called something like `url_parameters`. This indicates that any future parameters should be handled in `run` and just added to this argument.

This approach could then be supported in LaTeX output by simply giving `\sphinxcontribyoutube` a third argument to contain the url parameters.

Cheers,

David

akihironitta commented:
I'm considering utilising this extension in https://github.com/PyTorchLightning/pytorch-lightning/pull/7688. Any updates on this PR?

Robpol86 commented:
I'm currently working on another project but I plan on getting back to this some time next week.

Robpol86 committed:
Merge branch 'master' into timestamp_support

Robpol86 committed:
Switching from `:timestamp:` to `:url_parameters:`

Also adding LaTeX support.

Fixing a typo and linting around regex (raw string).

dham commented on README.rst:
```diff
@@ -31,14 +31,23 @@ parameters "aspect", "width", and "height" may optionally be provided::
 the premble. If they do not, then the default definition is used::
 
-    \newcommand{\sphinxcontribvimeo}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2}}\end{center}\end{quote}}
+    \newcommand{\sphinxcontribvimeo}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2#3}}\end{center}\end{quote}}
```
needs to be `[3]` to indicate that the command takes 3 parameters.

dham commented on sphinxcontrib/youtube/utils.py:
```diff
@@ -88,9 +89,9 @@ def visit_video_node_latex(self, node, platform, platform_url):
     if macro not in self.elements["preamble"]:
         self.elements["preamble"] += r"""
-        \newcommand{%s}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2}}\end{center}\end{quote}}
+        \newcommand{%s}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2#3}}\end{center}\end{quote}}
```
As above, needs to be `[3]`

dham reviewed (changes requested):
Thanks. I think this is basically good. I just found one tiny bug with parameter numbers.

Robpol86 committed:
Fix latex bug

Robpol86 commented:
Good catch. Fixed

i0bs commented:
@dham bumping.
